src/app/multi-editor/multi-editor.component.ts

```diff
@@ -160,16 +159,17 @@ export class MultiEditorComponent implements OnInit {
     return index;
   }
 
-  private addChecked(uuid: string) {
-    this.checkedRecords.includes(uuid) ? this.checkedRecords.splice(this.checkedRecords.indexOf(uuid), 1)
-      : this.checkedRecords.push(uuid);
+  private addOrRemoveChecked(value: boolean, uuid: string) {
+    this.checkedRecords[uuid] = value;
   }
 
   private selectOrDeselectAll(value: boolean) {
```

Comment:
this should fail ci, since you are calling `private` in the template.

Another idea is write setter and getter for `allSelected` instead of this function and you can use `[(ngModel)]=allSelected`